lib/kitchen/transport/base.rb

```diff
@@ -97,10 +105,43 @@ def close
         # @param command [String] command string to execute
         # @raise [TransportFailed] if the command does not exit successfully,
         #   which may vary by implementation
-        def execute(command) # rubocop:disable Lint/UnusedMethodArgument
+        def execute(command)
           raise ClientError, "#{self.class}#execute must be implemented"
         end
 
+        # Execute a command on the remote host and retry
+        #
+        # @param command [String] command string to execute
+        # @param retryable_exit_codes [Array] Array of exit codes to retry against
+        # @param max_retries [Fixnum] maximum number of retry attempts
+        # @param wait_time [Fixnum] number of seconds to wait before retrying command
+        # @raise [TransportFailed] if the command does not exit successfully,
+        #   which may vary by implementation
+        def execute_with_retry(command, retryable_exit_codes = [], max_retries, wait_time)
+          max_retries = 1 if max_retries.nil?
```

Comment:
So, I'm open to doing that, but I also wanted to guard against unexpected nils in the .kitchen.yml heirarchy that may have been overlooked/forgotten, but probably best to surface that in the .kitchen.yml / default_config.